be/src/column/column_helper.h

```diff
@@ -132,6 +132,14 @@ class ColumnHelper {
 
         return dst_column;
     }
+    static ColumnPtr duplicate_column(ColumnPtr src, MutableColumnPtr desc, UInt32Column::Ptr offsets) {
```

Comment:
duplicate a non-array column according to an array's offsets so that the non-array column aligns with the elements of an array to compute in vectorization.